appintro/src/main/java/com/github/appintro/internal/AppIntroViewPager.kt

```diff
@@ -158,7 +158,6 @@ internal class AppIntroViewPager(context: Context, attrs: AttributeSet) : ViewPa
                 if (isPermissionSlide && isSwipeForward(currentTouchDownX, event.x)) {
                     onNextPageRequestedListener?.onUserRequestedPermissionsDialog()
                 }
-                currentTouchDownX = event.x
```

Comment:
@cortinico This was added to prevent the next slide from being visible before permissions are granted. In your branch try this:
On slide 2 do a slow half backward swipe and then a rapid full forward swipe, a small portion of the next slide would be visible. In my testing I was actually able to bypass the permission check using this trick. So, .... Be careful while removing this.